Source/ui_unix/GamePad/GamePadDeviceListener_OSX.cpp

```diff
@@ -0,0 +1,350 @@
+#include "GamePadDeviceListener_OSX.h"
+#include "GamePadUtils.h"
+
+CGamePadDeviceListener::CGamePadDeviceListener(OnInputEvent OnInputEventCallBack, bool filter)
+    : OnInputEventCallBack(OnInputEventCallBack)
+    , m_running(true)
+    , m_filter(filter)
+{
+	m_thread = std::thread([this]() { InputDeviceListenerThread(); });
+}
+
+CGamePadDeviceListener::CGamePadDeviceListener(bool filter)
+    : CGamePadDeviceListener(nullptr, filter)
+{
+}
+
+CGamePadDeviceListener::~CGamePadDeviceListener()
+{
+	m_running = false;
+	CFRunLoopStop(CFRunLoopGetCurrent());
+	IOHIDManagerClose(m_hidManager, 0);
```

Comment:
The thread should be joined before closing the HID manager. I had random crashes here when quitting the app. Also, it seems `m_hidManager` could be made local to the `InputDeviceListenerThread` function. Would make sense to move the call there and remove the member from `CGamePadDeviceListener`.